pkg/supportbundle/spec.go

```diff
@@ -106,106 +106,200 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
-	if !util.IsHelmManaged() {
-		s, err := store.GetStore().GetRegistryDetailsForApp(app.GetID())
+		collectors, err := registry.UpdateCollectorSpecsWithRegistryData(supportBundle.Spec.Collectors, registrySettings, kotsKinds.Installation.Spec.KnownImages, kotsKinds.License, &kotsKinds.KotsApplication)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to get registry settings for app")
+			return nil, errors.Wrap(err, "failed to update collectors")
+		}
+		supportBundle.Spec.Collectors = collectors
+		b.Reset()
+		if err := s.Encode(supportBundle, &b); err != nil {
+			return nil, errors.Wrap(err, "failed to encode support bundle")
+		}
+		renderedSpec = b.Bytes()
+		secretName := GetSpecSecretName(app.GetSlug())
+		if key != "kotsadm" {
+			secretName = GetSpecSecretName(app.GetSlug() + "-" + key)
 		}
-		registrySettings = s
-	}
 
-	collectors, err := registry.UpdateCollectorSpecsWithRegistryData(supportBundle.Spec.Collectors, registrySettings, kotsKinds.Installation.Spec.KnownImages, kotsKinds.License, &kotsKinds.KotsApplication)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to update collectors")
-	}
-	supportBundle.Spec.Collectors = collectors
-	b.Reset()
-	if err := s.Encode(supportBundle, &b); err != nil {
-		return nil, errors.Wrap(err, "failed to encode support bundle")
-	}
-	renderedSpec = b.Bytes()
+		existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+		labels := kotstypes.MergeLabels(kotstypes.GetKotsadmLabels(), kotstypes.GetTroubleshootLabels())
+		if err != nil {
+			if kuberneteserrors.IsNotFound(err) {
+				secret := &corev1.Secret{
+					TypeMeta: metav1.TypeMeta{
+						APIVersion: "v1",
+						Kind:       "Secret",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      secretName,
+						Namespace: util.PodNamespace,
+						Labels:    labels,
+					},
+					Data: map[string][]byte{
+						SpecDataKey: renderedSpec,
+					},
+				}
 
-	secretName := GetSpecSecretName(app.GetSlug())
-	existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
-	labels := kotstypes.MergeLabels(kotstypes.GetKotsadmLabels(), kotstypes.GetTroubleshootLabels())
-	if err != nil {
-		if kuberneteserrors.IsNotFound(err) {
-			secret := &corev1.Secret{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: "v1",
-					Kind:       "Secret",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      secretName,
-					Namespace: util.PodNamespace,
-					Labels:    labels,
-				},
-				Data: map[string][]byte{
-					SpecDataKey: renderedSpec,
-				},
+				_, err = clientset.CoreV1().Secrets(util.PodNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to create support bundle secret")
+				}
+
+				logger.Debugf("created %q default support bundle spec secret", app.GetSlug())
+				// return supportBundle, nil
```

Comment:
This looks like it was a bug?  If so, let's just remove it.

Reply:
Thank you for point it out. It is not a bug in previous code, however, once I switched to use create multiple specs, it will cause early return. I have removed it.